test/src/main/scala/scalarules/test/srcjars/pack_sources_test.bzl

```diff
@@ -0,0 +1,91 @@
+load("@bazel_skylib//lib:unittest.bzl", "analysistest", "asserts")
+load("@bazel_skylib//lib:new_sets.bzl", "sets")
+
+# Tests and documents the functionality of phase_compile.bzl's _pack_source_jar
+
+def _single_source_jar_test_impl(ctx):
+    env = analysistest.begin(ctx)
+
+    target_under_test = analysistest.target_under_test(env)
+
+    srcjar_names = sets.make(
+        [j.basename for j in target_under_test[JavaInfo].source_jars],
+    )
+
+    # In line with Bazel's java_common.pack_sources,
+    # We return the initial .srcjar file since there are no source files
+    # Not sure where the second -src.jar comes from, maybe due to java rules
```

Comment:
Could you check if adding `expect_java_output = False` to `:source_jar` target makes this go away?

Reply:
It does! Thanks for the test case. I added one more to test multiple source jars, adding `SourceJar2.srcjar` to the test resources. Also refactored the test logic since it was insanely repetitive.